front/app/utils/containers/clickOutside/index.tsx

```diff
@@ -16,115 +18,100 @@ type Props = {
-    ) {
-      this.removeEventListeners();
-    }
-  }
-
-  componentWillUnmount() {
-    this.container = null;
-    this.removeEventListeners();
-  }
-
-  addEventListeners = () => {
-    if (this.props.closeOnClickOutsideEnabled) {
-      document.addEventListener('click', this.handle, true);
-      document.addEventListener('keyup', this.handle, true);
-    }
-  };
-
-  removeEventListeners = () => {
-    document.removeEventListener('click', this.handle, true);
-    document.removeEventListener('keyup', this.handle, true);
-  };
-
-  handle = (event) => {
-    // Press esc to close
-    if (event.type === 'keyup' && event.key === 'Escape') {
-      event.preventDefault();
-      this.props.onClickOutside(event);
-    }
-
-    if (
-      event.type === 'keyup' &&
-      event.key === 'Tab' &&
-      this.container &&
-      !this.container.contains(event.target)
-    ) {
-      this.props.onClickOutside(event);
+const ClickOutside = ({
+  children,
+  onClick,
+  onClickOutside,
+  onMouseEnter,
+  onMouseLeave,
+  onMouseDown,
+  className,
+  id,
+  setRef,
+  role,
+  closeOnClickOutsideEnabled = true,
+  isModal,
+  ariaLabelledBy,
+}: Props) => {
+  const containerRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    const handle = (event: Event) => {
+      // Press esc to close
+      if (event.type === 'keyup' && (event as KeyboardEvent).key === 'Escape') {
+        event.preventDefault();
+        onClickOutside(event);
+      }
+
+      if (
+        event.type === 'keyup' &&
+        (event as KeyboardEvent).key === 'Tab' &&
+        containerRef.current &&
+        !containerRef.current.contains(event.target as Node)
+      ) {
+        onClickOutside(event);
+      }
+
+      // Click outside to close
+      if (
+        event.type === 'click' &&
+        containerRef.current &&
+        !containerRef.current.contains(event.target as Node)
+      ) {
+        onClickOutside(event);
+      }
+    };
+
+    const addEventListeners = () => {
+      if (closeOnClickOutsideEnabled) {
+        document.addEventListener('click', handle, true);
+        document.addEventListener('keyup', handle, true);
+      }
+    };
+
+    const removeEventListeners = () => {
+      document.removeEventListener('click', handle, true);
+      document.removeEventListener('keyup', handle, true);
+    };
+
+    addEventListeners();
+    return () => removeEventListeners();
+  }, [closeOnClickOutsideEnabled, onClickOutside]);
+
+  useEffect(() => {
+    if (setRef && containerRef.current) {
+      setRef(containerRef.current);
     }
+  }, [setRef]);
 
-    // Click outside to close
-    if (
-      event.type === 'click' &&
-      this.container &&
-      !this.container.contains(event.target)
-    ) {
-      this.props.onClickOutside(event);
-    }
-  };
-
-  handleRef = (element: HTMLDivElement) => {
-    this.container = element;
-    this.props.setRef && this.props.setRef(element);
+  const handleOnMouseEnter = (event: MouseEvent) => {
```

Comment:
I think these handle functions are not really necessary- you can just pass `onMouseEnter` directly to `onMouseEnter` of the `Box`. Same for the other two handle functions below